### PR TITLE
add regional_annotation labels from annotation ETL as score to tile d…

### PR DESCRIPTION
**Pull request by druvpatel**
- added regional annotation labels from annotation ETL as scoring labels to tile df
- updated visualization code and jsons to support this
- small changes to better pick color of tiles for visualization

### CODE REVIEW

File: data_processing/pathology/common/preprocess.py [hunk 1]
```diff
@@ -209,6 +282,25 @@ def pretile_scoring(slide_file_path: str, output_dir: str, params: dict):
...
     df.loc[:, "purple_score"] = get_purple_scores (df['coordinates'], rbg_thumbnail,  thumbnail_tile_size)
 
+
+    # get pathology annotations for slide only if valid parameters
+    if slideviewer_dmt != None and slideviewer_dmt != "" and labelset != None and labelset != "":
+        annotation_url = os.path.join("http://", os.environ['MIND_API_URI'], "mind/api/v1/getPathologyAnnotation/", slideviewer_dmt, image_id, "regional", labelset)
+        logger.info(annotation_url)
+
+        response = urlopen(annotation_url)
+        response_text = response.read().decode(response.headers.get_content_charset())
```
Comment (aauker) [inline data_processing/pathology/common/preprocess.py:292]: If you feel up to it, using requests might simplify this code block:

```
response = requests.get(route)
annotation_geojson = response.json()
```

File: data_processing/pathology/common/preprocess.py [hunk 2]
```diff
@@ -21,9 +23,32 @@
...
-scoring_palette_as_list = [[int(x * 255) for x in scoring_palette.pop()] for i in range(NUM_COLORS)]
+from urllib.request import urlopen
+from shapely.geometry import shape, Point, Polygon
+
+from random import randint
+
+
+palette = sns.color_palette("viridis",as_cmap=True)
+categorical_colors = {}
+def get_tile_color(score):
```
Comment (aauker) [inline data_processing/pathology/common/preprocess.py:34]: Nice function!!!  There's a huge backlog of unit-tests, this is a good candidate too. I'll add these to the backlog ~

**Review by aauker — APPROVED** [review]
Awesome work.  Next steps might be to scale up and train a model!

File: data_processing/pathology/common/preprocess.py [hunk 3]
```diff
@@ -169,6 +240,8 @@ def pretile_scoring(slide_file_path: str, output_dir: str, params: dict):
 
     requested_tile_size       = params.get("tile_size")
     requested_magnification   = params.get("magnification")
+    slideviewer_dmt           = params.get("slideviewer_dmt")
```
[the remark this answers is not in the document]
Comment (druvpatel) [reply data_processing/pathology/common/preprocess.py:243]: Ooh! Didn't know you could specify a no-match return value this way, that's nice. will add!

File: data_processing/pathology/common/preprocess.py (hunk 1 above)
Comment (druvpatel) [reply data_processing/pathology/common/preprocess.py:292]: yes, this is way easier :)

**Comment by aauker** [comment]
Great PR. merge at will !

**Pull request merged**